### PR TITLE
Upgrade to Next.js 6

### PR DESCRIPTION
**Pull request by timneutkens**
Based on https://twitter.com/probablyup/status/1003857781894631424.

It was actually pretty easy, you weren't using `next/babel`, meaning Next.js's default preset wasn't used.
The other issue was that `preval` doesn't support Babel 7 yet, but this was fixable by using `DefinePlugin`.

### CODE REVIEW

**Comment by timneutkens** [comment]
Also add `.next` to the gitignore, since it doesn't have to be committed.

**Comment by timneutkens** [comment]
And moved `pages/App.js` and `pages/Home.js`, because they're not pages they shouldn't be inside the `pages` directory. If you keep them in `pages` they'll be generated as page bundles inside `.next` and will mess with the common modules calculation.

File: site/package.json [hunk 1]
```diff
@@ -15,30 +14,20 @@
     "styled-components": "^3.2.3"
   },
   "scripts": {
+    "dev": "next",
```
Comment (probablyup) [inline site/package.json]: Is like to keep it as just the “start” script for now.

Comment (timneutkens) [reply site/package.json]: Fixed

File: site/package.json [hunk 2]
```diff
@@ -3,7 +3,6 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "babel-plugin-preval": "^1.6.4",
     "babel-plugin-styled-components": "^1.5.1",
     "markdown-to-jsx": "^6.5.2",
     "next": "^5.0.0",
```
Comment (probablyup) [inline site/package.json]: Doesn’t look like v6 was upgraded here? That’s probably why the PR Netlify deploy is failing.

Comment (timneutkens) [reply site/package.json]: Yep, for some reason `yarn` crashes on node-gyp 🤔

**Comment by probablyup** [comment]
@timneutkens If you're using Node 10 locally, I think Yarn locally [needs to be >= 1.7.0](https://github.com/yarnpkg/yarn/releases/tag/v1.7.0) to be compatible.

**Comment by probablyup** [comment]
Thanks Tim!

**Pull request merged**